parser/encoding/request.go

```diff
@@ -0,0 +1,138 @@
+package encoding
+
+import (
+	"github.com/cockroachdb/errors"
+
+	"encr.dev/parser"
+	v1 "encr.dev/proto/encore/parser/schema/v1"
+)
+
+// ParamLocation is the request/response home of the parameter
+type ParamLocation int16
```

Comment:
This can probably be just `int`; the efficiency doesn't really matter and `int16` tricks you into thinking the size matters